syntax = "proto3";

service Todo
{
    rpc createTodo(TodoItem) returns (TodoItem) {}
    rpc readTodos(NoParams) returns (TodoItems) {}
    rpc readTodoStream(NoParams) returns (TodoItems) {}
}

message NoParams {}

message TodoItem
{
    int32 id = 1;
    string text = 2;
}

message TodoItems
{
    repeated TodoItem items = 1;
}